migrator/src/main/java/com/redhat/lightblue/migrator/Utils.java

```diff
@@ -68,68 +79,109 @@ public static LightblueClient getLightblueClient(String configPath)
-                compareDocs(inconsistentPaths, sourceArray.get(x), destinationArray.get(x), path,excludes);
+    /**
+     * This stupidity is required because data types of the source and
+     * dest might be different, but they might have the same
+     * value. Like, a number represented as string in one system could
+     * be an int in another. That is valid, but JSON comparison fails
+     * for that, so we check if the two nodes are *really* different,
+     * meaning: for anything but dates, check string equivalence. For
+     * dates, normalize by TZ and check.
+     */
+    private static boolean reallyDifferent(Object source,Object dest) {
+        if(source==null)
+            if(dest==null)
+                return false;
+            else
+                return true;
+        else if(dest==null)
+            return true;
+        else {
+            String s1=source.toString();
+            String s2=dest.toString();
+            if(s1.equals(s2))
+                return false;
+
+            // They are different strings
+            // Do they look like dates?
+            Date d1;
+            Date d2;
+            DateFormat fmt=ClientConstants.getDateFormat();
+            try {
+                d1=fmt.parse(s1);
+            } catch (Exception e) {
+                return true;
             }
-        } else if (JsonNodeType.OBJECT.equals(sourceDocument.getNodeType())) {
-            if (!JsonNodeType.OBJECT.equals(destinationDocument.getNodeType())) {
-                inconsistentPaths.add(StringUtils.isEmpty(path) ? "*" : path);
-                return;
+            try {
+                d2=fmt.parse(s2);
+            } catch (Exception e) {
+                return true;
             }
+            return d1.getTime()!=d2.getTime();
+        }
+    }
 
-            // compare object contents
-            Iterator<Map.Entry<String, JsonNode>> itr = sourceDocument.fields();
-
-            while (itr.hasNext()) {
-                Map.Entry<String, JsonNode> entry = itr.next();
+    private static boolean isExcluded(List<String> exclusions,String path) {
+        if(!exclusions.contains(path)) {
+            for(String x:exclusions)
+                if(matches(path,x))
+                    return true;
+        } else
+            return true;
+        return false;
+    }
 
-                compareDocs(inconsistentPaths, entry.getValue(), destinationDocument.get(entry.getKey()),
-                            StringUtils.isEmpty(path) ? entry.getKey() : path + "." + entry.getKey(),excludes);
+    private static boolean matches(String path,String pattern) {
+        String[] patternComponents=pattern.split("\\.");
+        String[] pathComponents=path.split("\\.");
+        if(patternComponents.length<=pathComponents.length) {
+            for(int i=0;i<patternComponents.length;i++) {
+                if(pathComponents[i].equals(patternComponents[i])||
+                   "*".equals(patternComponents[i]))
+                    ;
```

Comment:
Looks like this if statement doesn't do anything. Should the conditional be reversed?